jerry-core/mem/mem-poolman.cpp

```diff
@@ -430,12 +436,16 @@ mem_pools_alloc_longpath (void)
   JERRY_ASSERT (MEM_POOL_SIZE <= mem_heap_get_chunked_block_data_size ());
   JERRY_ASSERT (MEM_POOL_CHUNKS_NUMBER >= 1);
 
+  MEM_HEAP_VALGRIND_MEMPOOL_REQUEST_START ();
   mem_pool_chunk_t *pool_start_p = (mem_pool_chunk_t*) mem_heap_alloc_chunked_block (MEM_HEAP_ALLOC_LONG_TERM);
```

Comment:
The allocation could invoke GC that can free heap block. The free wouldn't be accounted, as the request appears to be performed from pool allocator. For valgrind this appears as memory leak:

 LEAK SUMMARY:
    definitely lost: 256 bytes in 1 blocks
 ...

Example code:
// make VALGRIND=ON debug.linux
//  a26c45421988d138a01408ecfefb80816e3bcf81 with merged 8f44fa70253a9270ff4b2cd58be12b91004087b4
// valgrind ./build/bin/debug.linux/jerry test.js
// allocates RegExp object that stores byte-code on heap and is freed by GC during a call to `f`
```js
new RegExp ("foo", undefined);

function f()
{
  function g()
  {
  }

  return g;
}

for (i = 0; i < 100; i++)
{
  f ();
}
```